setup.py

```diff
@@ -174,154 +80,31 @@ def detect_encoding(readline):
-        output_dir = os.path.join(self.build_dir, arch)
-        try:
-            binaries = []
-            for bin, srcs in self.scripts.items():
-                # Only compile srcs with c files
-                try:
-                    if not any([src.endswith(".c") for src in srcs]):
-                        continue
-                except AttributeError:
-                    if not src.endswith(".c"):
-                        continue
-
-                objs = compiler.compile(srcs, output_dir=output_dir,
-                                        include_dirs=include_dirnames,
-                                        extra_postargs=extra_postargs,
-                                        debug=False)
-
-                bin_path = os.path.join(output_dir, bin)
-
-                compiler.link_executable(objs, bin_path,
-                                         library_dirs=library_dirnames)
-                binaries.append(bin_path)
-
-            # Replace dict script with actual before build_scripts.run() call
-            self.scripts = binaries
-        except AttributeError:  # Not a dict
-            pass
-
-        print("##################################################")
-
-        build_scripts.run(self)  # Call actual script
-
-        # If success, remove script build dir
-        if os.path.isdir(output_dir):
-            print("Removing script build dir: %s" % output_dir)
-            rmtree(output_dir)
-
-
-def make_gnulib():
-    print(">> Not a Linux system: configuring and making Gnulib libraries...")
-
-    libfilename = "%s/libgnu.a" % GNULIB_LIB_DIR
-    if os.path.isfile(libfilename):
-        print(">> Found libgnu.a... skipping configure and make")
-    else:
-        commands = ["./configure", "make"]
-        for command in commands:
-            print(">> %s" % command)
-            try:
-                check_call(command, cwd=GNULIB_BUILD_DIR)
-            except CalledProcessError:
-                raise InstallationError("Error compiling Gnulib")
-
-    if not os.path.isfile(libfilename):
-        raise InstallationError("Expected to find: %s" % libfilename)
+if "HDF5_DIR" in os.environ:
+    hdf5_dir = os.environ["HDF5_DIR"]
+    include_dirnames.append(os.path.join(hdf5_dir, "include"))
+    library_dirnames.append(os.path.join(hdf5_dir, "lib"))
 
-    to_rm = ["%s/getopt.h" % GNULIB_LIB_DIR]
-    for filename in to_rm:
-        if os.path.isfile(filename):
-            print(">> Removing: %s" % filename)
-            os.remove(filename)
+load_data_module = Extension('_load_data_c_ext', # needs to match C file PyInit definition
+                            sources=source_files,
+                            include_dirs = include_dirnames,
+                            libraries = libs,
+                            library_dirs = library_dirnames,
+                            define_macros = c_define_macros,
+                            extra_compile_args = ["-UNDEBUG"] # NB: Keep assert macros functioning
```

Comment:
Replace with `#undef NDEBUG` in C file.

Reply:
I cannot find in a manual page or reference anywhere of what takes precedence for the C/C++ Preprocessor. The command line or a value set inline. I would intuitively believe it would always be inline directives to take precedence. I can attempt the change to see if it works. It just seems potentially a bit more fragile.

Reply:
Yeah I think the arguments for the command-line are just things that are set at the beginning of running the C preprocessor. I think the command-line approach is more fragile, especially since we don't completely control what command line gets run.

Reply:
OK I will include